src/pybind/mgr/cephadm/module.py

```diff
@@ -509,6 +511,12 @@ def notify(self, notify_type: str, notify_id: Optional[str]) -> None:
                 self._kick_serve_loop()
         if notify_type == "pg_summary":
             self._trigger_osd_removal()
+        if notify_type == 'mgr_map':
+            old_prometheus_scrape_list = self.prometheus_scrape_list
+            new_prometheus_scrape_list = self.prometheus_service.get_mgr_scrape_list()
+            if old_prometheus_scrape_list != new_prometheus_scrape_list:
+                self.prometheus_scrape_list = new_prometheus_scrape_list
+                self._service_action('redeploy', 'prometheus')
```

Comment:
I suspect there is a potential race between a MGR failover and enabling mgr/prometheus if we're not called for  **all** mgr_map updates.